adlik_serving/apis/model_operate_impl.cc

```diff
@@ -155,10 +156,47 @@ tensorflow::Status ModelOperateImpl::deleteModelVersion(const ModelOperateReques
     FailureResponse(rsp, modelName + " model version delete failure");
     return status;
   }
-  ROLE(StorageLoop).once();
-  ROLE(BoardingLoop).once();
+  update();
   SuccessResponse(rsp);
   return status;
 }
+
+tensorflow::Status ModelOperateImpl::activateModel(const ModelOperateRequest& req, ModelOperateResponse& rsp) {
+  std::string modelName = req.model_name();
+  std::string modelVersion = req.model_version();
+  INFO_LOG << "activate model " << modelName << ", version " << modelVersion;
+  tensorflow::Status status = tensorflow::Status::OK();
+  auto config = ROLE(ModelStore).find(modelName);
+  if (!config) {
+    INFO_LOG << modelName << " model is not exist";
+    FailureResponse(rsp, modelName + " model is not exist");
+    return status;
+  }
+  std::string modelPath = config->getBasePath();
+  VersionDirectoryState versionDirectoryState;
+  cub::filesystem().children(modelPath, versionDirectoryState);
+  if (!versionDirectoryState.contains(stoi(modelVersion))) {
+    FailureResponse(rsp, modelName + " model version " + modelVersion + " is not exist");
+    return status;
+  }
+  VersionPolicyProto originVersionPolicy(config->version_policy());
+  VersionPolicyProto newVersionPolicy;
+  newVersionPolicy.mutable_specific()->add_versions(stoi(modelVersion));
+  ROLE(ModelStore).updatePolicy(modelName, newVersionPolicy);
+  update();
+  if (!ROLE(ManagedStore).exist(modelName) || !ROLE(ManagedStore).isNormal(modelName)) {
```

Comment:
The correct process should be: first load the new model, if load success then unload the old model